src/core/modules/Star.hpp

```diff
@@ -243,6 +244,179 @@ struct Star
 		bS *= 1000./ plx;
 		bS += diffPos * MAS2RAD * 1000.;
    }
+
+   inline void getBinaryOrbit(double epoch, Vec3d& v) const {
+      double pmra = 0.;
+      double pmdec = 0.;
+      double rv = 0.;
+      double plx = 0.;
+      double ra = 0.;
+      double dec = 0.;
+      getBinaryOrbit(epoch, v, ra, dec, plx, pmra, pmdec, rv);
+   }
+
```

Comment:
Can you please add some literature documentation here, where you found the algorithms? Maybe Explanatory Supplement (which ed), some other book dedicated on double stars? Or did you do all the math yourself from scratch?

Reply:
I end up just do most of the math myself from definitions of these orbital elements. From the definitions, one can compute the 2D orbit of the secondary star in the reference frame of the primary and then rotate the 2D plane to 3D orbit as seen on the sky and then apply proper motion to the barycenter of the binary star. No fancy algorithms used here.

Reply:
Well, maybe not really fancy and can be done by oneself, but I had expected you followed some textbook, so congratulations.

Reply:
Ah ok, let me google to see if there is any good textbook on those. I forget which resources I use as I indeed have looked up what those orbital elements mean exactly.